AIPscan/typesense_helpers.py

```diff
@@ -0,0 +1,287 @@
+        except Exception:
+            pass
+
+        # Create collection
+        fields = collection_fields_from_model(model)
+
+        ts_client.collections.create(
+            {"name": collection_prefix(table), "fields": fields}
+        )
+
+
+def model_instance_to_document(model, instance, model_fields=None):
+    if model_fields is None:
+        model_fields = collection_fields_from_model(model)
+
+    document = {}
+
+    for field in model_fields:
+        if hasattr(instance, field["name"]):
+            value = getattr(instance, field["name"])
+
+            if field["name"] == "id":
+                value = str(value)
+
+            if isinstance(value, datetime.datetime):
+                value = int(value.timestamp())
+
+            if isinstance(value, FileType):
+                value = str(value).replace("FileType.", "")
+
+            if value is not None:
+                document[field["name"]] = value
+
+    return document
+
+
+def augment_document_with_aip_data(table, document, aip_cache, result):
+    if table == "aip":
+        document["original_file_count"] = aip_cache["original_file_counts"][result.id]
+
+    if table == "file":
+        document["storage_service_id"] = aip_cache["storage_service_ids"][result.aip_id]
+        document["storage_location_id"] = aip_cache["storage_location_ids"][
+            result.aip_id
+        ]
+
+        # Round AIP create date to the day
+        dt = aip_cache["create_dates"][result.aip_id]
+        dt = datetime.datetime(*dt.timetuple()[0:3])
+        document["aip_create_date"] = int(dt.strftime("%s"))
+
+
+def populate_index(update_threshold_percent=1):
+    ts_client = client()
+
+    # Cache AIP storage services/locations
+    aip_cache = {}
+    aip_cache["storage_service_ids"] = {}
+    aip_cache["storage_location_ids"] = {}
+    aip_cache["create_dates"] = {}
+    aip_cache["original_file_counts"] = {}
+
+    results = AIP.query.all()
+
+    for result in results:
+        aip_cache["storage_service_ids"][result.id] = result.storage_service_id
+        aip_cache["storage_location_ids"][result.id] = result.storage_location_id
+        aip_cache["create_dates"][result.id] = result.create_date
+        aip_cache["original_file_counts"][result.id] = result.original_file_count
+
+    # Add documents to collections
+    for model in APP_MODELS:
+        table = get_model_table(model)
+        model_fields = collection_fields_from_model(model)
+
+        results = model.query.all()
+
+        docs = []
+        indexed = 0
+        completed_percent = 0
+
+        for result in results:
+            document = model_instance_to_document(model, result, model_fields)
+            augment_document_with_aip_data(table, document, aip_cache, result)
+
+            docs.append(document)
+            perform_bulk_document_creation(ts_client, table, docs)
+
+            indexed += 1
+            completed_percent = math.ceil((indexed / len(results)) * 100)
```

Comment:
I think it's clearer without the parenthesis around `indexed / len(results)`.

Reply:
Fixed!